### PR TITLE
docs(transformer): add documentation for `optional-catch-binding` plugin

### PR DESCRIPTION
**Pull request by Dunqing**
follow-up #4881

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#5065** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5065?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5064** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5064?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_transformer/src/es2019/optional_catch_binding.rs [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+//! This plugin transform catch clause without parameter to add a parameter called `unused` in catch clause.
+//!
+//! > This plugin is included in `preset-env`, in ES2019
+//!
+//! ## Example
+//!
+//! Input:
+//! ```js
+//! try {
+//!  throw 0;
```
Comment (overlookmotel) [inline crates/oxc_transformer/src/es2019/optional_catch_binding.rs:12]:
```suggestion
//!   throw 0;
```

**Review by overlookmotel — CHANGES_REQUESTED** [review]

File: crates/oxc_transformer/src/es2019/optional_catch_binding.rs [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+//! } catch (_unused) {
```
Comment (overlookmotel) [inline crates/oxc_transformer/src/es2019/optional_catch_binding.rs:13]:
```suggestion
//! } catch {
```

**Review by overlookmotel — CHANGES_REQUESTED** [review]
Maybe remove the duplicate comment lower down (lines 54-63).

Or, if leaving it, correct the typo in that comment - "too" should be "to" at line 59.

**Comment by Dunqing** [comment]
> Maybe remove the duplicate comment lower down (lines 54-63).

Removed

File: crates/oxc_transformer/src/es2019/optional_catch_binding.rs [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+//!
+//! > This plugin is included in `preset-env`, in ES2019
+//!
+//! ## Example
+//!
+//! Input:
+//! ```js
+//! try {
+//!   throw 0;
+//! } catch () {
```
Comment (overlookmotel) [inline crates/oxc_transformer/src/es2019/optional_catch_binding.rs:13]: I don't think `catch ()` is legal syntax.

```suggestion
//! } catch {
```

**Review by overlookmotel — CHANGES_REQUESTED** [review]

**Review by overlookmotel — APPROVED** [review]

**Pull request merged**